okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/DnsOverHttps.kt

```diff
@@ -247,7 +254,12 @@ class DnsOverHttps internal constructor(
       val query = DnsRecordCodec.encodeQuery(hostname, type)
 
       if (post) {
-        url(url).post(query.toRequestBody(DNS_MESSAGE))
+        url(url)
+          .cacheUrlOverride(
+            url.newBuilder()
+              .addQueryParameter("hostname", hostname).build(),
```

Comment:
Ooooh it does now!

Reply:
This is supercool

Reply:
I'll take a look.

Reply:
> I'll take a look.

Thank you so much.

Reply:
> I'll take a look.

Any update? Thanks

Reply:
You are right about the cache miss in that method.

That method is there, and works for get, in order to avoid a conditional get when we already have a result, delaying resolution.

I'm tempted to live with that performance regression, but open for discussions. So that PR simplifies the code to remove that.

https://github.com/square/okhttp/pull/8383

Reply:
The tests probably won't fail because of the change, unless I calculate a delay past the heuristic cache checks, so I should follow up with an additional test in that PR.

Reply:
> You are right about the cache miss in `getCacheOnlyResponse`.
> 
> The `getCacheOnlyResponse` method is there, and works correctly for get, in order to avoid a conditional network request when we already have a result, delaying resolution. I'll double check this from PR history.
> 
> I'm tempted to live with that performance regression, but open for discussions. So this PR simplifies the code to remove that.
> 
> #8383

Thanks for your explanation. However, I don't think `getCacheOnlyResponse` could help avoiding ‘unnecessary’ conditional request. Firstly, if `CacheStrategy` figures out that we need a conditional request to verify that our cache is still available, I think we should send one. And, what a `onlyIfCached` CacheControl can do is to enforce a response with code HTTP_GATEWAY_TIMEOUT when a conditional request(or any network request) is wanted,(see `CacheStrategy#compute()`) so that we would still abandon the cache and make a request given current implementation.

Reply:
> Firstly, if CacheStrategy figures out that we need a conditional request to verify that our cache is still available, I think we should send one

Yes, that's the conclusion I think I came to. Please leave comments on that PR if there is anything to change further.

BTW thanks for the really thorough analysis here, while it was technically working, it's much simpler code now, and I obviously didn't understand the flow as well as I thought I did. But digging in like you have to find issues and improve the logic, is really valuable. So, thanks.

Reply:
> > Firstly, if CacheStrategy figures out that we need a conditional request to verify that our cache is still available, I think we should send one
> 
> Yes, that's the conclusion I think I came to. Please leave comments on that PR if there is anything to change further.
> 
> BTW thanks for the really thorough analysis here, while it was technically working, it's much simpler code now, and I obviously didn't understand the flow as well as I thought I did. But digging in like you have to find issues and improve the logic, is really valuable. So, thanks.

Off the topic: thanks for your contribution to okhttp. And you are my role model. Your enthusiasm in programming, pretty readable code with high performance inspires me a lot. Thank you.